internal/util/cephcmds.go

```diff
@@ -86,21 +87,24 @@ func getPools(ctx context.Context, monitors string, cr *Credentials) ([]cephStor
 	return pools, nil
 }
 
-// GetPoolID searches a list of pools in a cluster and returns the ID of the pool that matches
-// the passed in poolName parameter
-func GetPoolID(ctx context.Context, monitors string, cr *Credentials, poolName string) (int64, error) {
-	pools, err := getPools(ctx, monitors, cr)
+// GetPoolID fetches the ID of the pool that matches the passed in poolName
+// parameter
+func GetPoolID(monitors string, cr *Credentials, poolName string) (int64, error) {
+	conn, err := connPool.Get(poolName, monitors, cr.KeyFile)
```

Comment:
seems like arguments are not passed in the proper order can you please double-check? https://github.com/ceph/ceph-csi/blob/master/internal/util/conn_pool.go#L123

Reply:
right, good catch! the pool is not needed for `Get()`, but the userId.